src/ixdat/techniques/deconvolution.py

```diff
@@ -0,0 +1,254 @@
+            ))
+
+        return t_sig, partial_current
+
+
+    def extract_kernel(self, signal_name, cutoff_pot=0, tspan=None, t_bg=None):
+        """Extracts a Kernel object from a measurement.
+
+        Args:
+            signal_name (str): Signal name from which the kernel/impule
+                response is to be extracted.
+            cutoff_pot (int): Potential which the defines the onset of the
+                impulse. Must be larger than the resting potential before the
+                impulse.
+            tspan(list): Timespan from which the kernel/impulse response is
+                extracted.
+            t_bg (list): Timespan that corresponds to the background signal.
+        """
+        x_curr, y_curr = self.get_current(tspan=tspan)
+        x_pot, y_pot = self.get_potential(tspan=tspan)
+        x_sig, y_sig = self.get_signal(signal_name, tspan=tspan, t_bg=t_bg)
+
+        if signal_name == 'M32':
+            t0 = x_curr[np.argmax(y_pot > cutoff_pot)] #time of impulse
+        elif signal_name == 'M2':
+            t0 = x_curr[np.argmax(y_pot < cutoff_pot)]
+        else:
+            print('mass not found')
+
+        x_sig = x_sig - t0
+
+        y_sig = y_sig[x_sig>0]
+        x_sig = x_sig[x_sig>0]
+
+        y_curr = y_curr[x_curr>t0]
+        x_curr = x_curr[x_curr>t0]
+        y_pot = y_pot[x_pot>t0]
+        x_pot = x_pot[x_pot>t0]
+
+        kernel = Kernel(
+            MS_data = np.array([x_sig, y_sig]),
+            EC_data = np.array([x_curr, y_curr, x_pot, y_pot])
+            )
+
+
+        return kernel
+
+
+
+class Kernel:
+    """
+    Kernel class implementing different functionalities for datatreatment of
+    kernel/impulse response data.
+    """
+    def __init__ (
+        self,
+        parameters = {},
+        MS_data = None,
+        EC_data = None,
+    ):
+        """Initializes a Kernel object either in functional form by defining the
+        mass transport parameters or in the measured form by passing of EC-MS
+        data.
+
+        Args:
+            parameters (dict): Dictionary containing the mass transport
+                parameters with the following keys:
+                    D: Diffusion constant in liquid
+                    L: Working distance between electrode and gas/liq interface
+                    V: Gas sampling volume of the chip
+                    V_dot: Volumetric capillary flow
+                    kH: Dimensionless Henry volatility
+                MS_data (list): List of numpy arrays containing the MS signal
+                    data.
+                EC_data (list): List of numpy arrays containing the EC (time,
+                    current, potential).
+        """
+
+        if MS_data is not None and parameters:
+            raise Exception(
+                'Kernel can only be initialized with data OR parameters, not both'
+                )
+        if EC_data is not None and MS_data is not None:
```

Comment:
I wonder if this kind of switching in the `__init__` means there are really two things we call `Kernel` that share some features but are different classes. 
Maybe there should be a `MeasuredKernel` and a `CalculatedKernel` that both inherit from `Kernel`. 
Do I understand right that a measured kernel is something you would use if you've done an impulse response experiment, and thus have data that should correspond to a kernel?
At the very least, the docstring needs to go into more detail of when you'd use parameters and when you'd use data.

Reply:
Yes, that's correct! I like you're suggestion with Kernel inheriting from Measurement, I added that as a TODO. For now I'll just leave it like it is to get the repo for the paper up and running.

Reply:
Great. I can't find the TODO though.